main/read.c

```diff
@@ -397,27 +397,27 @@ static boolean setSourceFileName (vString *const fileName)
  *   Line directive parsing
  */
 
-static int skipWhite (void)
+static void skipWhite (char **str)
 {
-	int c;
-	do
-		c = mio_getc (File.mio);
-	while (c == ' '  ||  c == '\t');
-	return c;
+	while (**str == ' '  ||  **str == '\t')
+		(*str)++;
 }
 
-static unsigned long readLineNumber (void)
+static unsigned long readLineNumber (char **str)
 {
+	char *s;
 	unsigned long lNum = 0;
-	int c = skipWhite ();
-	while (c != EOF  &&  isdigit (c))
+
+	skipWhite (str);
+	s = *str;
+	while (*s != '\0' && isdigit (*s))
 	{
-		lNum = (lNum * 10) + (c - '0');
-		c = mio_getc (File.mio);
+		lNum = (lNum * 10) + (*s - '0');
+		s++;
 	}
-	mio_ungetc (File.mio, c);
```

Comment:
I didn't quite understand why they did the ungetc() here as the character gets tested below. It should behave the same because this is followed by readLineNumber() which skips spaces at the beginning. This way just looks more correct to me.